WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -412,31 +412,15 @@ def time_until_competition
 
   def calculate_dues
     country_iso2 = Country.find_by(id: params[:country_id])&.iso2
-    country_band = CountryBand.find_by(iso2: country_iso2)&.number
-
-    update_exchange_rates_if_needed
-    input_money_us_dollars = Money.new(params[:entry_fee_cents].to_i, params[:currency_code]).exchange_to("USD").amount
-
-    registration_fee_dues_us_dollars = input_money_us_dollars * CountryBand::PERCENT_REGISTRATION_FEE_USED_FOR_DUE_AMOUNT
-    country_band_dues_us_dollars = CountryBand::BANDS[country_band][:value] if country_band.present? && country_band > 0
+    dues_per_competitor_in_usd = DuesCalculator.dues_per_competitor_in_usd(country_iso2, params[:base_entry_fee_lowest_denomination], params[:currency_code])
 
     # times 100 because later currency conversions require lowest currency subunit, which is cents for USD
-    price_per_competitor_us_cents = [registration_fee_dues_us_dollars, country_band_dues_us_dollars].compact.max * 100
-
-    if ActiveRecord::Type::Boolean.new.cast(params[:competitor_limit_enabled])
-      estimated_dues_us_cents = price_per_competitor_us_cents * params[:competitor_limit].to_i
-      estimated_dues = Money.new(estimated_dues_us_cents, "USD").exchange_to(params[:currency_code]).format
-
-      render json: {
-        dues_value: estimated_dues,
-      }
-    else
-      price_per_competitor = Money.new(price_per_competitor_us_cents, "USD").exchange_to(params[:currency_code]).format
-
-      render json: {
-        dues_value: price_per_competitor,
-      }
-    end
+    price_per_competitor_us_cents = dues_per_competitor_in_usd * 100
```

Comment:
Rather than doing arbitrary multiplications with a "magic" 100 number, you could pass around the Ruby `Money` object instead.

Reply:
But don't we need to convert to cent? I'm not sure exactly why, but we used to multiply by 100 earlier too. Is there any way to do with Ruby Money here?

Reply:
You can have the `DuesCalculator.dues_per_competitor_in_usd` method return a RubyMoney object. That has the additional advantage of making it obvious that the return value is USD. You can rename the method to `DuesCalculator.dues_per_competitor` and the return object itself knows (and can report to the user if necessary) that it is a money object representing USD.

Since Ruby is a dynamic language, you can do shenanigans like:
```ruby
  ruby_money = Money.new(200, "USD") # 2 USD because 200 means "US cents"
  competitor_limit = 50 # read from request body or whatever

  estimated_total_dues = ruby_money * competitor_limit # yes, you can multiply a RubyMoney object with an integer

  puts estimated_total_dues.format # expected output: 100 USD 
```